sklearn/linear_model/_glm/tests/test_glm.py

```diff
@@ -1108,3 +1108,190 @@ def test_newton_solver_verbosity(capsys, verbose):
             "The inner solver detected a pointwise Hessian with many negative values"
             " and resorts to lbfgs instead." in captured.out
         )
+
+
+def test_lbfgs_convergence_error():
+
+    def _load_edgecase():
+        """Dataset to cause a failing test in GLM model."""
+        X = [
```

Comment:
I just realized that there is another edge case which the current fix is not able to catch, if we run

```
X = X[:,-1:]
```

(i.e., only use the last row)

then the models converge with

```
lbfgs
intercept: 3.896592058397603
coefs:    [0.]

# newton-cholesky
intercept:  3.2885810888136406
coefs:     [0.00087977]
```

Reply:
Hum, that is bad. Does this still happen even with tiny values of `tol` (while adjusting `max_iter` as needed)?

Please open a dedicated issue for this one if tiny `tol` values do not fix it.

Reply:
What are the values of the Poisson loss for both models at convergence? Could you add a print statement to also display the gradient value at the last iterate? Maybe there are below machine precision level and there is nothing we can do about it.

Reply:
On such a low parameter space, one could also disply the 2x2 Hessian matrix of the Newton Cholesky method at each iterate, and it's condition number.